app/src/main/java/at/bitfire/icsdroid/ProcessEventsTask.kt

```diff
@@ -77,10 +79,16 @@ class ProcessEventsTask(
             Log.d(Constants.TAG, "Removing all alarms from ${uid}: $this")
             alarms.clear()
         }
-        subscription.defaultAlarmMinutes?.let { minutes ->
+        val isAllDay = DateUtils.isDate(dtStart)
+        val alarmMinutes = if (isAllDay)
+            subscription.defaultAllDayAlarmMinutes
+        else
+            subscription.defaultAlarmMinutes
+        if (alarmMinutes != null) {
             // Check if already added alarm
             val alarm = alarms.find { it.description.value.contains("*added by ICSx5") }
-            if (alarm != null) return@let
+            if (alarm != null) return@apply
```

Comment:
Thanks, looks good!

@ArnyminerZ One question: is this code still necessary, i.e. can it ever be called? I think it's only called for Events that are fetched from the server and thus can never contain an ICSx5 alarm, right? Also I couldn't find the "*added by ICSx5" string somewhere else, so I guess it's not set somewhere else.

In this case we could remove those lines.

Reply:
If I'm not wrong, that piece is for not adding multiple times the same alarm, otherwise, each time the event gets processed a new alarm is added. By marking it with the `*added by ICSx5` tag we make sure that no duplicates are created.

Reply:
But where is this "*added by ICSx5" added?

And are events not created from scratch every time?

I think this could be a relict from a previous version

Reply:
Will review again, once this is resolved :)

Reply:
> But where is this "*added by ICSx5" added?
> 
> And are events not created from scratch every time?
> 
> I think this could be a relict from a previous version

Mmmmh... I'll take a look asap

Reply:
Yup, that's true, seems to be from somewhere in the past, it's no longer used. I've now removed it 😉